packages/manager/apps/web/client/app/private-database/private-database.service.js

```diff
@@ -485,62 +484,24 @@ export default class PrivateDatabase {
     });
   }
 
-  listAvailableRamDurations(serviceName, opts) {
-    return this.canOrderRam(serviceName).then((canOrderRam) => {
-      let rtn;
-
-      if (canOrderRam) {
-        rtn = this.$http
-          .get(`${this.swsProxypassOrderPath}/${serviceName}/ram`, {
-            params: { ram: opts.ram },
-          })
-          .then((response) => response.data || [])
-          .catch(() => []);
-      }
-
-      return rtn;
-    });
-  }
-
-  getRamPrices(serviceName, opts) {
-    const durationsTab = [];
-    const defer = this.$q.defer();
-
-    this.listAvailableRamDurations(serviceName, opts)
-      .then((durations) => {
-        defer.notify(durations);
-
-        return this.$q.all(
-          map(durations, (duration) =>
-            this.$http
-              .get(
-                `${this.swsProxypassOrderPath}/${serviceName}/ram/${duration}`,
-                {
-                  params: {
-                    ram: opts.ram,
-                  },
-                },
-              )
-              .then((durationDetails) => {
-                const details = angular.copy(durationDetails.data);
-                details.duration = duration;
-                durationsTab.push(details);
-                defer.notify(durationsTab);
-              }),
-          ),
-        );
+  getRamPlan(serviceName, planCode) {
+    return this.$http
+      .get(`${this.swsProxypassUpgradePath}/${serviceName}/${planCode}`, {
+        params: {
+          quantity: 1,
+        },
       })
-      .then(() => defer.resolve(durationsTab))
-      .catch(() => defer.resolve(durationsTab));
-    return defer.promise;
+      .then((response) => response.data.order);
```

Comment:
```suggestion
      .then(({data: {order}}) => order);
```
use destructed object instead